### PR TITLE
Add per partition replication lag metrics.

### PR DESCRIPTION
**Pull request by zzmao**
Allow non-clusterMap node to get replication lag from metadata exchange.
Add per partition replication lag for VCR backup.

### CODE REVIEW

**Comment by jsjtzyy** [comment]
Can this PR be extended to normal data nodes?  It seems very useful if the number of metrics won't be a  problem. I feel like it could be applied to monitor replication progress during migration.

**Comment by zzmao** [comment]
I feel it would be some stress for ingraph, let me check if it can be configurable.

**Comment by zzmao** [comment]
Config added so the metric can be enabled per host for normal node, which is usually what we want to observe.

File: ambry-replication/src/main/java/com.github.ambry.replication/ReplicationMetrics.java [hunk 1]
```diff
@@ -406,6 +410,15 @@ public void incrementInvalidMessageError(PartitionId partitionId) {
     }
   }
 
+  public void addMetricsForPartition(PartitionId partitionId) {
+    if (partitionLags.put(partitionId, new HashMap<>()) == null) {
```
Comment (lightningrob) [inline ambry-replication/src/main/java/com.github.ambry.replication/ReplicationMetrics.java]: This overwrite doesn't look correct.

File: ambry-replication/src/main/java/com.github.ambry.replication/ReplicationMetrics.java [hunk 2]
```diff
@@ -406,6 +410,15 @@ public void incrementInvalidMessageError(PartitionId partitionId) {
     }
   }
 
+  public void addMetricsForPartition(PartitionId partitionId) {
```
Comment (lightningrob) [inline ambry-replication/src/main/java/com.github.ambry.replication/ReplicationMetrics.java]: Need javadoc

**Comment by zzmao** [comment]
@lightningrob  Thought about your suggestion last night. Store lag per partition in Helix is possible and it is friendly to our metric system.  @jsjtzyy 
But this needs some effort(figure out how persist value in helix), so I would like to have this PR first and get an overview about the backup progress.
Since the metrics are configurable, we can turn it off any time. Meanwhile, we can check how big impact is it to our metric system.

File: ambry-replication/src/main/java/com.github.ambry.replication/ReplicaThread.java [hunk 3]
```diff
@@ -879,6 +879,8 @@ private void writeMessagesToLocalStoreAndAdvanceTokens(List<ExchangeMetadataResp
               totalBlobsFixed += messageInfoList.size();
               remoteReplicaInfo.setToken(exchangeMetadataResponse.remoteToken);
               remoteReplicaInfo.setLocalLagFromRemoteInBytes(exchangeMetadataResponse.localLagFromRemoteInBytes);
+              replicationMetrics.updateLagMetricForRemoteReplica(remoteReplicaInfo,
```
Comment (lightningrob) [inline ambry-replication/src/main/java/com.github.ambry.replication/ReplicaThread.java]: Can we update the replication test to verify that the new metrics are getting populated?

Comment (zzmao) [reply ambry-replication/src/main/java/com.github.ambry.replication/ReplicaThread.java]: Test added.

**Pull request merged**